src/ray/raylet/lineage_cache.cc

```diff
@@ -187,6 +187,12 @@ void LineageCache::AddWaitingTask(const Task &task, const Lineage &uncommitted_l
 void LineageCache::AddReadyTask(const Task &task) {
   auto new_entry = LineageEntry(task, GcsStatus_UNCOMMITTED_READY);
   RAY_CHECK(lineage_.SetEntry(std::move(new_entry)));
+  // Add the task to the cache of tasks that may be flushed.
+  uncommitted_ready_tasks_.insert(task.GetTaskSpecification().TaskId());
+
+  // Try to flush the task to the GCS.
+  // TODO(swang): Allow a pluggable policy for when to flush.
+  RAY_CHECK_OK(Flush());
```

Comment:
So this will result in flushing the lineage cache very frequently, right? E.g., every time a task becomes ready, right?

You already have a TODO, but in the future, we may want to remove this and just call `Flush` on a timer, right?

Reply:
Yes.